### PR TITLE
Move LmdbDatabase and MerkleDatabase to libsawtooth

### PR DESCRIPTION
**Pull request by agunde406**
Also moves required traits and structs to move them.

### CODE REVIEW

File: libsawtooth/Cargo.toml [hunk 1]
```diff
@@ -54,8 +55,11 @@ lmdb-store = ["lmdb-zero", "stores"]
 receipt-store = ["stores"]
 redis-store = ["redis", "stores"]
 stores = []
-validator-internals = ["hex", "openssl", "lmdb-zero"]
+validator-internals = ["cbor-codec","hex", "openssl", "lmdb-zero"]
```
Comment (ltseeley) [inline libsawtooth/Cargo.toml:58]: Missing a space here (also, lmdb-zero isn't in its alphabetical spot).

File: libsawtooth/src/state/merkle.rs [hunk 2]
```diff
@@ -678,7 +676,7 @@ fn encode_and_hash(node: Node) -> Result<(Vec<u8>, Vec<u8>), StateDatabaseError>
 /// Given a path, split it into its parent's path and the specific branch for
 /// this path, such that the following assertion is true:
 ///
-/// ```
+/// ```ignore
```
Comment (ltseeley) [inline libsawtooth/src/state/merkle.rs:679]: This isn't captured in the commit message; why is this necessary?

File: libsawtooth/Cargo.toml (hunk 1 above)
Comment (agunde406) [reply libsawtooth/Cargo.toml]: fixed

File: libsawtooth/src/state/merkle.rs (hunk 2 above)
Comment (agunde406) [reply libsawtooth/src/state/merkle.rs:679]: added to commit, this test was not being running in its old location but when moved was being ran. But the doc test fails, so I ignored it.
